server/website/website/tasks/async_tasks.py

```diff
@@ -1206,6 +1125,9 @@ def map_workload(map_workload_input):
 
     scores = {}
     for workload_id, workload_entry in list(workload_data.items()):
+        LOG.info('%s: %s', newest_result.workload.pk, workload_id)
+        if newest_result.workload.pk == workload_id:
+            continue
```

Comment:
I check the same workload here.

Reply:
I see. But why not check if it's the same workload above when looping through the unique workloads? That way you don't spend time processing the same workload data that will never be used.

Reply:
You are right.